From f325434b9fa3ca0a55a58947777c3fd825f590e9 Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Tue, 26 Aug 2014 17:19:05 -0400
Subject: [input] alps: report 2 touches when we've > 2 fingers

Message-id: <1409073549-31548-15-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 89248
O-Subject: [PATCH RHEL7 14/18] Input: alps - report 2 touches when we've > 2 fingers
Bugzilla: 1107819
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1107819
Upstream: since v3.17
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7882179
Tested by me on a Toshiba Portege R30 with a v7 Alps touchpad

commit 99d9996c5c3b47c14c8717c7dd4692a25b83b3f7
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Jul 25 22:48:02 2014 -0700

    Input: alps - report 2 touches when we've > 2 fingers

    If we detect more then 2 fingers report 2 touches, rather then only
    reporting the upper left corner of the bounding box.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/input/mouse/alps.c b/drivers/input/mouse/alps.c
index f16fe7c..5026600 100644
--- a/drivers/input/mouse/alps.c
+++ b/drivers/input/mouse/alps.c
@@ -454,7 +454,7 @@ static void alps_report_semi_mt_data(struct psmouse *psmouse, int fingers)
   fingers = f->pressure > 0 ? 1 : 0;
  }
 
- alps_report_mt_data(psmouse, (fingers <= 2) ? fingers : 1);
+ alps_report_mt_data(psmouse, (fingers <= 2) ? fingers : 2);
 
  input_mt_report_finger_count(dev, fingers);
 
-- 
1.7.1